### PR TITLE
ci: introduce buildkite

### PR DESCRIPTION
**Pull request by galak**
Introduce setup to utilize buildkite for CI purposes.

### CODE REVIEW

**Comment by nashif** [comment]
i see a kite..

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# Save off where we started so we can go back there
+WORKDIR=${PWD}
+
+if [ -n "${BUILDKITE_PULL_REQUEST_BASE_BRANCH}" ]; then
+   git fetch -v origin ${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
+   git checkout FETCH_HEAD
+   git config --local user.email "builds@zephyrproject.org"
+   git config --local user.name "Zephyr CI"
+   git merge --no-edit "${BUILDKITE_COMMIT}" || {
```
Comment (carlescufi) [inline .buildkite/hooks/pre-command:14]: `git rebase`?

Comment (galak) [reply .buildkite/hooks/pre-command:14]: I want to leave this as is, I think it mimics how github and other CI systems do things.  Thus having the git rebase in scripts/ci/run_ci.sh covers our needs.

Comment (carlescufi) [reply .buildkite/hooks/pre-command:14]: but then what's the  point of having this  merge here?

Comment (galak) [reply .buildkite/hooks/pre-command:14]: The intent of the merge is to have the git repo look similar to how other CI systems setup things.  So any scripts or such we develop look similar regardless of what CI system we happen to use.

Comment (galak) [reply .buildkite/hooks/pre-command:14]: W/o this change the `git rebase` in `scripts/ci/run_ci.sh` doesnt do much as the git tree will just be whatever the PR is and thus you need the merge to pull master into it.

**Review by nashif — APPROVED** [review]

File: .buildkite/hooks/pre-command (hunk 1 above)
Comment (nashif) [reply .buildkite/hooks/pre-command:14]: the rebase originally was done to deal with linters, not sure if it still make sense with sanitycheck. However, one main advantage we might have with rebasing is that everytime we run, we build against latest master and not against whatever commit the PR was originally created against...

**Review by carlescufi — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**